### PR TITLE
Make everything compile cleanly with -Wpedantic.

### PR DESCRIPTION
**Pull request by AWoloszyn**
This makes SPIRV_WARN_EVERYTHING compile cleanly.

### CODE REVIEW

**Comment by AWoloszyn** [comment]
Fixes: https://github.com/KhronosGroup/SPIRV-Tools/issues/153

File: test/UnitSPIRV.h [hunk 1]
```diff
@@ -215,5 +215,19 @@ inline std::string MakeLongUTF8String(size_t num_4_byte_chars) {
   return result;
 }
 
+template <typename T>
+std::string DefaultParameterPrinter(
```
Comment (dekimir) [inline test/UnitSPIRV.h]: Needs blurb.

File: test/UnitSPIRV.h [hunk 2]
```diff
@@ -215,5 +215,19 @@ inline std::string MakeLongUTF8String(size_t num_4_byte_chars) {
...
+    const typename testing::TestParamInfo<T>& instance) {
+  std::ostringstream ss;
+  ss << instance.index;
+  return ss.str();
+}
+
+template<typename T>
+std::string EnumParameterPrinter(
```
Comment (dekimir) [inline test/UnitSPIRV.h]: Needs blurb.

File: test/Validate.Capability.cpp [hunk 3]
```diff
@@ -51,8 +53,15 @@ using testing::ValuesIn;
 using ValidateCapability =
     spvtest::ValidateBase<tuple<string, pair<string, vector<string>>>>;
 
+std::string ValidateCapabilityPrinter(
```
Comment (dekimir) [inline test/Validate.Capability.cpp]: Needs blurb.

File: test/AssemblyContext.cpp [hunk 4]
```diff
@@ -82,7 +83,8 @@ INSTANTIATE_TEST_CASE_P(
       // A very long string, encoded after an initial word.
       // SPIR-V limits strings to 65535 characters.
       {std::string(65535, 'a'), {1}},
-    }));
+    }),
+    &DefaultParameterPrinter<EncodeStringCase>);
```
Comment (dekimir) [inline test/AssemblyContext.cpp]: This is onerous (we must add junk to all INSTANTIATE calls now and forever) and surprising (we can't use INSTANTIATE the way Googletest doc suggest). There has to be a better way. :pray:

Comment (dekimir) [inline test/AssemblyContext.cpp]: Turns out the variable-arguments list may be empty.  This compiles cleanly with `--pedantic`:

```
#define foo(a, ...) a
int main() {
  return foo(0,);
}
```

**Comment by AWoloszyn** [comment]
Replaced arguments with ,

**Comment by dneto0** [comment]
+2

**Pull request merged**